docs/CIP/CIP-XXXX/messages/client/subscribe.md

```diff
@@ -0,0 +1,201 @@
+    },
+    "topics": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "blockchain": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string"
+              },
+              "network": {
+                "type": "string"
+              }
+            },
+            "required": ["name", "network"]
+          },
+          "signature": {
+            "type": "string",
+            "pattern": "^[A-Za-z0-9+/=]*$"
+          },
+          /* <blockchain specific grouped fields> */
+          "cardano": {
+            "type": "object",
+            "properties": {
+              "credentials": {
+                "type": "object",
+                "properties": {
+                  "payment": {
+                    "type": "array",
+                    "items": {
+                      "type": "string",
+                      "pattern": "^[A-Za-z0-9+/=]*$"
+                    }
+                  },
+                  "stake": {
+                    "type": "array",
+                    "items": {
+                      "type": "string",
+                      "pattern": "^[A-Za-z0-9+/=]*$"
+                    }
+                  }
+                },
+                "required": ["payment", "stake"]
+              },
+              "extensions": {
+                "type": "object",
+                "properties": {
+                  "resolveTxInput": {
+                    "type": "boolean",
+                    "default": false
+                  },
+                  "assetMetadata": {
+                    "type": "boolean",
+                    "default": false
+                  }
+                }
+              }
+            },
+            "required": ["credentials", "extensions"]
+          }
+        },
+        "required": ["blockchain", "signature"]
+      }
+    },
+    "timestamp": {
+      "type": "string",
+      "format": "date-time"
+    }
+  },
+  "required": ["type", "topics", "timestamp"],
+  "additionalProperties": false
+}
+```
+
+The `signature` field is generated by signing a SHA256 HMAC hashed string. The prehash string is constructed by concatenating the subscribed `blockchains` + `timestamp`.
+
+> [!NOTE]
+> For subscribing to multiple blockchains, the prehash string must be generated for each as `<blockchain.name>.<blockchain.network>`, for example:
+> `cardano.mainnet+2024-06-27T12:34:56Z`.
+
+Apply a SHA256 HMAC using either the payment signing key or staking signing key, and then base64-encode it as final payload within the initial message. Pass the output of this SHA256 HMAC to the signature field.
+
+The signature verifies ownership of the private key, whose corresponding public key is used to filter relevant block transactions.
+
+## Subscribe Example
+
+```json
+{
+  "type": "subscribe",
+  "topics": [
+    {
+      "blockchain": { "name": "cardano", "network": "mainnet" },
```

Comment:
I just don't see why we would have a dedicated connection for let's say a chain like Bitcoin which may publish new Txs for a client once every 10 minutes.

Reply:
Just because an open connection is almost for free, while

- handling and writing each field
- writing, deploying and maintaining a mux / demux service

require human work: the most expensive resource in our context.